test_ros2trace/test/test_ros2trace/test_trace.py

```diff
@@ -199,37 +203,52 @@ def run_trace_command_start(
         if wait_for_start:
             assert process.stdout
             stdout = ''
-            while 'press enter to stop...' not in stdout:
+            time_start = time.time()
+            time_now = time_start
+
+            def timedout(now: float) -> bool:
+                return timeout is not None and now - time_start > timeout
+
+            while 'press enter to stop...' not in stdout and not timedout(time_now):
                 stdout += process.stdout.read(1)
+                time_now = time.time()
```

Comment:
maybe a bit simpler?

```suggestion
            time_start = time.time()

            def timedout() -> bool:
                return timeout is not None and time.time() - time_start > timeout

            while 'press enter to stop...' not in stdout and not timedout():
                stdout += process.stdout.read(1)
```